python/ppml/example/fgboost_regression/fgboost-tutorial.md

```diff
@@ -0,0 +1,172 @@
+
+The code is available in projects, including [Client 1 code](fgboost_regression_party_1.py) and [Client 2 code](fgboost_regression_party_2.py). You could directly start two different terminals are run them respectively to start a federated learning, and the order of start does not matter. Following is the detailed step-by-step tutorial to introduce how the code works.
+
+### 2.1 FL Context
+First, import the package and initilize FL Context.
+```python
+from bigdl.ppml.fl import *
+from bigdl.ppml.fl.algorithms.fgboost_regression import FGBoostRegression
+import pandas as pd
+
+init_fl_context()
+```
+
+### 2.2 Private Set Intersection
+Then, read the data,
+
+Party 1:
+```python
+df_train = pd.read_csv('house-prices-train-1.csv')
+```
+
+Party 2:
+```python
+df_train = pd.read_csv('house-prices-train-2.csv')
+```
+
+To get the data intersection which the 2 parties can do federated learning, we have to the Private Set Intersection (PSI) algorithm first.
+```python
+from bigdl.ppml.fl.algorithms.psi import PSI
+ids = df_train['Id']
+psi = PSI()
+salt = psi.get_salt()
+psi.upload_set(ids, salt)
+intersection = psi.download_intersection()
+df_train = df_train.ix(intersection) # select the intersection part of training data
+```
+### 2.3 Data Preprocessing
+We provide a `preprocess` method in the code, including normalization and one-hot. The output is a DataFrame with all numerical features. You can check the detail in the [code]()
+
+```python
+df_train = preprocess(df_train)
+```
+### 2.4 FGBoost Instance
+Create the FGBoost instance `FGBoostRegression` with default parameters 
+```
+fgboost_regression = FGBoostRegression()
+```
+### 2.5 Training
+Then call `fit` method to train
+
+Party 1:
+```python
+# party 1 does not own label, so directly pass all the features
+fgboost_regression.fit(df_train, feature_columns=df_train.columns, num_round=100)
+```
+Party 2:
+```python
+# party 2 owns label, so pass the features and the labels
+fgboost_regression.fit(df_x, df_y, feature_columns=df_x.columns, label_columns=['SalePrice'], num_round=100)
+```
+### 2.6 Save & Load
+Save the model and load it back
+
+Party 1:
+```python
+fgboost_regression.save_model('/tmp/fgboost_model_1.json')
+loaded = FGBoostRegression.load_model('/tmp/fgboost_model_1.json')
+```
+Party 2:
+```python
+fgboost_regression.save_model('/tmp/fgboost_model_2.json')
+loaded = FGBoostRegression.load_model('/tmp/fgboost_model_2.json')
+```
+### 2.7 Prediction
+Party 1:
+```python
+df_test = pd.read_csv('house-prices-test-1')
+result = fgboost_regression.predict(df_test, feature_columns=df_test.columns)
+```
+
+Party 2:
+```python
+df_test = pd.read_csv('house-prices-test-2')
+result = fgboost_regression.predict(df_test, feature_columns=df_test.columns)
+```
+## 3 Run FGBoost
+FL Server is required before running any federated applications. Check [Start FL Server]() section for details.
```

Comment:
running the FL Server and Clients on the same node?

Reply:
Currently, yes. If no, we have to assume some URL like `192.168.0.1` and set `clientTarget` to it, and user need to do some similar modification before they run this example.